# HG changeset patch
# User Max Brister <max@2bass.com>
# Date 1352071736 25200
#      Sun Nov 04 16:28:56 2012 -0700
# Node ID 6e780fb15c14586afaf56f4328b29d04828845ce
# Parent  a7b22144318a895ca048aa98e99d01f5907f6dfd
Make Vdebug_on_interrupt not be static

* sighandlers.cc: Make Vdebug_on_interrupt not be static.
* sighandlers.h: Declare Vdebug_on_interrupt.

* pt-jit.cc: Include sighandlers.h.

diff --git a/libinterp/interp-core/pt-jit.cc b/libinterp/interp-core/pt-jit.cc
--- a/libinterp/interp-core/pt-jit.cc
+++ b/libinterp/interp-core/pt-jit.cc
@@ -27,16 +27,17 @@ along with Octave; see the file COPYING.
 #include <config.h>
 #endif
 
 #include "debug.h"
 #include "defun.h"
 #include "ov.h"
 #include "pt-all.h"
 #include "pt-jit.h"
+#include "sighandlers.h"
 #include "symtab.h"
 #include "variables.h"
 
 static bool Venable_jit_debugging = false;
 
 static bool Venable_jit_compiler = true;
 
 #ifdef HAVE_LLVM
diff --git a/libinterp/interpfcn/sighandlers.cc b/libinterp/interpfcn/sighandlers.cc
--- a/libinterp/interpfcn/sighandlers.cc
+++ b/libinterp/interpfcn/sighandlers.cc
@@ -56,17 +56,17 @@ along with Octave; see the file COPYING.
 // Nonzero means we have already printed a message for this series of
 // SIGPIPES.  We assume that the writer will eventually give up.
 int pipe_handler_error_count = 0;
 
 // TRUE means we can be interrupted.
 bool can_interrupt = false;
 
 // TRUE means we should try to enter the debugger on SIGINT.
-static bool Vdebug_on_interrupt = false;
+bool Vdebug_on_interrupt = false;
 
 // Allow users to avoid writing octave-workspace for SIGHUP (sent by
 // closing gnome-terminal, for example).  Note that this variable has
 // no effect if Vcrash_dumps_octave_core is FALSE.
 static bool Vsighup_dumps_octave_core = true;
 
 // Similar to Vsighup_dumps_octave_core, but for SIGTERM signal.
 static bool Vsigterm_dumps_octave_core = true;
diff --git a/libinterp/interpfcn/sighandlers.h b/libinterp/interpfcn/sighandlers.h
--- a/libinterp/interpfcn/sighandlers.h
+++ b/libinterp/interpfcn/sighandlers.h
@@ -169,9 +169,12 @@ private:
 
   static bool instance_ok (void);
 
   static octave_child_list_rep *instance;
 
   static void cleanup_instance (void) { delete instance; instance = 0; }
 };
 
+// TRUE means we should try to enter the debugger on SIGINT.
+extern OCTINTERP_API bool Vdebug_on_interrupt;
+
 #endif
